src/core/Object3D.js

```diff
@@ -357,6 +357,7 @@ Object3D.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
 			this.children.push( object );
 
 			object.dispatchEvent( _addedEvent );
+			this.dispatchEvent( { type: 'childadded', child: object } );
```

Comment:
From https://github.com/mrdoob/three.js/pull/11459#issuecomment-398116192.

I think it's okay creating an object per event however I also understand the concern for devs how perform many add and remove operations that would like to see a single object. 

Since the listeners in `dispatchEvent()` are processed in a sync fashion, I think it would be safe to have single event objects and just change the `child` reference. What do you think?

Reply:
I think it's good - we just have to set and then remove the objects from the cached event before and after dispatch. I'll merge this as is and then make a new PR with this.